sql/core/src/main/scala/org/apache/spark/sql/execution/command/views.scala

```diff
@@ -152,10 +152,10 @@ case class CreateViewCommand(
             sparkSession,
             analyzedPlan,
             aliasedPlan.schema,
-            originalText,
-            child))
+            originalText))
       } else {
-        aliasedPlan
+        assert(isTemporary)
+        View(None, isTemporary, aliasedPlan)
```

Comment:
It's a bit weird to create the `View` wrapper at view creation time. Shall we do it in `SessionCatalog.lookupTempView`?